File: 219-contains-duplicate-II/solution2.cpp
// Runtime: 136ms, Beats 46.63% of users with C++
// Memory: 82.92MB, Beats 12.36% of users with C++

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int n=nums.size();
        set<int> s;
        int i=0;
        for (int j=0;j<n;j++)
        {
            if(j-i>k)
            {
                s.erase(nums[i]);
                i++;
            }
            if (s.find(nums[j])!=s.end()) return true;
            s.insert(nums[j]);
        }
        return false;
    }
};

File: 977-squares-of-a-sorted-array/solution2.py
# Runtime: 150ms, Beats 83.29% of users with Python3
# Memory: 18.49MB, Beats 92.99% of users with Python3

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        return sorted([x**2 for x in nums])
            